tests/onnxruntime/test_modeling.py

```diff
@@ -251,6 +254,58 @@ def test_model_on_cpu_str(self):
         self.assertEqual(model.device, cpu)
         self.assertListEqual(model.providers, ["CPUExecutionProvider"])
 
+    def test_missing_execution_provider(self):
+        with self.assertRaises(ValueError) as cm:
+            model = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider="ThisProviderDoesNotExist")
+
+        self.assertTrue("but the available execution providers" in str(cm.exception))
+
+        is_onnxruntime_gpu_installed = (
+            subprocess.run("pip list | grep onnxruntime-gpu", shell=True, capture_output=True).stdout.decode("utf-8")
+            != ""
+        )
+        is_onnxruntime_installed = "onnxruntime " in subprocess.run(
+            "pip list | grep onnxruntime", shell=True, capture_output=True
+        ).stdout.decode("utf-8")
+
+        if not is_onnxruntime_gpu_installed:
+            for provider in ["CUDAExecutionProvider", "TensorrtExecutionProvider"]:
+                with self.assertRaises(ImportError) as cm:
+                    _ = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider=provider)
+
+                self.assertTrue(
+                    f"Asked to use {provider}, but `onnxruntime-gpu` package was not found." in str(cm.exception)
+                )
+        else:
+            logger.info("Skipping CUDAExecutionProvider/TensorrtExecutionProvider without `onnxruntime-gpu` test")
+
+        # need to install first onnxruntime-gpu, then onnxruntime for this test to pass,
+        # thus overwritting onnxruntime/capi/_ld_preload.py
+        if is_onnxruntime_installed and is_onnxruntime_gpu_installed:
+            for provider in ["CUDAExecutionProvider", "TensorrtExecutionProvider"]:
+                with self.assertRaises(ImportError) as cm:
+                    _ = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider=provider)
+
+                self.assertTrue(
+                    "`onnxruntime-gpu` is installed, but GPU dependencies are not loaded." in str(cm.exception)
+                )
+        else:
+            logger.info("Skipping double onnxruntime + onnxruntime-gpu install test")
+
+        # LD_LIBRARY_PATH can't be set at runtime,
+        # see https://stackoverflow.com/questions/856116/changing-ld-library-path-at-runtime-for-ctypes
+        # testing only for TensorRT as having ORT_CUDA_UNAVAILABLE is hard
+        if is_onnxruntime_gpu_installed:
+            for provider in ["TensorrtExecutionProvider"]:
+                out = subprocess.run(
+                    f"CUDA_PATH='' LD_LIBRARY_PATH='' python tests/onnxruntime/load_model.py {provider}",
```

Comment:
What is the goal of this test? And where comes load_model.py?

Reply:
Added the missing load_model.py, thanks. This is to test that a meaningful error message is raised in case TRT requirements can not be loaded.

Reply:
Do we really put it in a separate script as it is just a one-liner?

Reply:
right, will edit!

Reply:
Should be all good now.